jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/Main.java

```diff
@@ -66,10 +70,27 @@ public static Map<String, Object> shipData(Map<String, String> params) {
         ESManager.configureIndexAndTypes(ds,errorList);
         errorList.addAll(new EntityManager().uploadEntityData(ds));
         ExternalPolicies.getInstance().uploadPolicyDefinition(ds);
-        if("aws".equals(ds)) {
-        	errorList.addAll(new AssetGroupStatsCollector().collectAssetGroupStats());
-        	errorList.addAll(new IssueCountManager().populateViolationsCount());
-        	errorList.addAll(new AssetsCountManager().populateAssetCount());
+        try {
+            DatasourceData datasourceData = DatasourceDataFetcher.getInstance().fetchDatasourceData(ds);
+            if (datasourceData != null) {
+                List<String> accountIds = datasourceData.getAccountIds();
+                Map<String, List<String>> assetGroupDomains = datasourceData.getAssetGroupDomains();
+
+                if (assetGroupDomains != null && !assetGroupDomains.isEmpty()) {
+                    AssetGroupStatsCollector assetGroupStatsCollector = new AssetGroupStatsCollector();
+                    errorList.addAll(assetGroupStatsCollector.collectAssetGroupStats(datasourceData));
+                }
+
+                if (accountIds != null && !accountIds.isEmpty()) {
+                    IssueCountManager issueCountManager = new IssueCountManager();
+                    errorList.addAll(issueCountManager.populateViolationsCount(ds, accountIds));
+
+                    AssetsCountManager assetsCountManager = new AssetsCountManager();
+                    errorList.addAll(assetsCountManager.populateAssetCount(ds, accountIds));
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.error("Error while updating stats", e);
```

Comment:
Let's have this error added to `errorList` as well

Reply:
because if we don't do this - failed count with 0 errors will be considered as a success. Right?

Reply:
yes, I have added now